python/tests/utils.py

```diff
@@ -223,17 +223,28 @@ def econf_foreach_cluster(econf, fn, name='cluster_httpbin_default'):
             break
     assert found_cluster
 
-def assert_valid_envoy_config(config_dict, v2=False):
-    with tempfile.NamedTemporaryFile() as temp:
-        temp.write(bytes(json.dumps(config_dict), encoding = 'utf-8'))
-        temp.flush()
-        f_name = temp.name
-        cmd = [ENVOY_PATH, '--config-path', f_name, '--mode', 'validate']
+def assert_valid_envoy_config(config_dict, extra_dirs=[], v2=False):
```

Comment:
This was one of the items on my list that I wanted us to address. I like the fact that this is now not dependent on local installs. However, the part I still do not like which we can address in the future is that its mixed into the "unit" test which I found unexpected when trying to develop.